code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/message/InAppMessagePresentable.kt

```diff
@@ -169,32 +170,28 @@ internal class InAppMessagePresentable(
         return InAppMessageWebViewClient(
             inAppMessage.settings,
             presentationUtilityProvider
-        ) { url -> handleInAppUrl(url) }
+        ) { url -> handleInAppUri(url) }
     }
 
     /**
-     * Handles the in-app url. Does so by first checking if the component that created this message
-     * is able to handle the url.
-     * @param url the url to handle
+     * Handles the in-app uri. Does so by first checking if the component that created this message
+     * is able to handle the uri.
+     * @param uri the uri to handle
      * @return true if the url was handled internally by the web-view client, false otherwise
      */
-    private fun handleInAppUrl(url: String): Boolean {
-        // Check if the component that created this message is able to handle the url
-        val handledByListener =
-            inAppMessage.eventListener.onUrlLoading(this@InAppMessagePresentable, url)
-
-        // Check if this URL can be opened by the URLOpening
-        val handled = handledByListener || if (InAppMessageWebViewClient.isValidUrl(url)) {
-            val uriOpened = presentationUtilityProvider.openUri(url)
-            if (uriOpened) {
-                presentationDelegate?.onContentLoaded(
-                    this@InAppMessagePresentable,
-                    PresentationListener.PresentationContent.UrlContent(url)
-                )
-            }
-            true
-        } else {
-            false
+    @VisibleForTesting
+    internal fun handleInAppUri(uri: String): Boolean {
+        // First check if the component that created this message is able to handle the uri.
+        // Otherwise check if this URI can be opened by the utility provider via UriOpening service
+        val handled = inAppMessage.eventListener.onUrlLoading(this@InAppMessagePresentable, uri) ||
+            presentationUtilityProvider.openUri(uri)
```

Comment:
Does 2.x only notify delegate method if `UriService` is able to open the URI? Wondering if there is a case to also notify the delegate about malformed URI

Reply:
In 2.x we notify if the [url is loaded successfully](https://github.com/adobe/aepsdk-core-android/blob/main-v2/code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewClient.java#L162).